docs/pipelines.md

```diff
@@ -465,6 +465,141 @@ There are a lot of scenarios where `WhenExpressions` can be really useful. Some
+
+**Note:** Scoping `when` expressions to a `Task` and its dependent `Tasks` is deprecated. To guard a `Task` and its 
+dependent `Tasks`, cascade `when` expressions to the specific dependent `Tasks` to be guarded as well.
+
+```
+                                     tests
+                                       |
+                                       v
+                                 manual-approval
+                                 |            |
+                                 v        (approver)
+                            build-image       |
+                                |             v
+                                v          slack-msg
+                            deploy-image
+```
+
+Taking the use case above, a user who wants to guard `manual-approval` and its dependent `Tasks` can design the 
+`Pipeline` as such:
+
+```yaml
+tasks:
+...
+- name: manual-approval
+  runAfter:
+    - integration-tests
+  when:
+    - input: $(params.git-action)
+      operator: in
+      values:
+        - merge
+  taskRef:
+    name: manual-approval
+
+- name: slack-msg
+  when:
+    - input: $(params.git-action)
+      operator: in
+      values:
+        - merge
+  params:
+    - name: approver
+      value: $(tasks.manual-approval.results.approver)
+  taskRef:
+    name: slack-msg
+
+- name: build-image
+  when:
+    - input: $(params.git-action)
+      operator: in
+      values:
+        - merge
+  runAfter:
+    - manual-approval
+  taskRef:
+    name: build-image
+
+- name: deploy-image
+  when:
+    - input: $(params.git-action)
+      operator: in
+      values:
+        - merge
+  runAfter:
+    - build-image
+  taskRef:
+    name: deploy-image
+```  
+
+#### Guarding a `Task` only
+
+To guard a `Task` only and unblock execution of its dependent `Tasks`, set the global default scope of `when` expressions
+to `Task` using the `scope-when-expressions-to-task` field in [`config/config-feature-flags.yaml`](install.md#customizing-the-pipelines-controller-behavior)
+by changing it to "true".
+
+```
+                                     tests
+                                       |
+                                       v
+                                 manual-approval
+                                 |            |
+                                 v        (approver)
+                            build-image       |
+                                |             v
+                                v          slack-msg
+                            deploy-image
+```
+
+Taking the use case above, a user who wants to guard `manual-approval` only can design the `Pipeline` as such:
+
+```yaml
+tasks:
+...
+- name: manual-approval
+  runAfter:
+    - tests
+  when:
+    - input: $(params.git-action)
+      operator: in
+      values:
+        - merge
+  taskRef:
+    name: manual-approval
+
+- name: slack-msg
+  params:
+    - name: approver
+      value: $(tasks.manual-approval.results.approver)
```

Comment:
hmmm similar to my comment above, I'm not sure how this would be possible? even tho skipping is scoped to only `manual-approval`, my understanding is that this should still get skipped b/c otherwise what value will ` $(tasks.manual-approval.results.approver)` have?

if this IS how this is working right now, I'm guessing this is an empty string? [This design doc has some info about why this might not be what we want](https://docs.google.com/document/d/1tV1LgPOINnmlDV-oSNdLB39IlLcQRGaYAxYZjVwVWcs/edit), particularly [Option b (results default)](https://docs.google.com/document/d/1tV1LgPOINnmlDV-oSNdLB39IlLcQRGaYAxYZjVwVWcs/edit#heading=h.phrhie8akizi), which also links to [a previous discussion about this](https://docs.google.com/document/d/10iEJqVstY6k3KNvAXgffIJLcHRbPQ-GIAfQk5Dlrf3c/edit#heading=h.19ybq5czkwi6)

The summary section of https://github.com/tektoncd/community/pull/240 uses this same Approver example, pointing out that maybe using a string like "None" might be the desired behavior. For example maybe the slack message is something like "Pull request 4555 was approved by <approver>" - Something like "Pull request 4555 was approved by None" or maybe ("Pull request 4555 was approved implicitly; no manual approval needed") would be better than "Pull request 4555 was approved by "

Reply:
> hmmm similar to my comment above, I'm not sure how this would be possible? even tho skipping is scoped to only `manual-approval`, my understanding is that this should still get skipped b/c otherwise what value will ` $(tasks.manual-approval.results.approver)` have?
> 

$(tasks.manual-approval.results.approver) has NO value i.e. no empty string if manual-approval is skipped. The task result resolution indicates the value is missing and 
slack-msg has no sufficient data to run and will be skipped. But at least the slack-msg was scheduled and attempted without creating any taskRun.  I think this aligns with the proposal.

> if this IS how this is working right now, I'm guessing this is an empty string? [This design doc has some info about why this might not be what we want](https://docs.google.com/document/d/1tV1LgPOINnmlDV-oSNdLB39IlLcQRGaYAxYZjVwVWcs/edit), particularly [Option b (results default)](https://docs.google.com/document/d/1tV1LgPOINnmlDV-oSNdLB39IlLcQRGaYAxYZjVwVWcs/edit#heading=h.phrhie8akizi), which also links to [a previous discussion about this](https://docs.google.com/document/d/10iEJqVstY6k3KNvAXgffIJLcHRbPQ-GIAfQk5Dlrf3c/edit#heading=h.19ybq5czkwi6)
> 
> The summary section of [tektoncd/community#240](https://github.com/tektoncd/community/pull/240) uses this same Approver example, pointing out that maybe using a string like "None" might be the desired behavior. For example maybe the slack message is something like "Pull request 4555 was approved by " - Something like "Pull request 4555 was approved by None" or maybe ("Pull request 4555 was approved implicitly; no manual approval needed") would be better than "Pull request 4555 was approved by "

Once we add support for the task results without results, the message will be sent saying PR was approved implicitly or by no one.

Reply:
yes, and by skipping it right now because of missing results (instead of failing) sets it up for continued execution once default results are available and the user enables them